Increase priority for global bucket merges

vekterli opened the pull request:
@geirst @toregge please review

To avoid global buckets competing with (and usually being starved by)
default bucket space merges, explicitly prioritize default bucket
merges above most other load.

Increases both distributor-internal maintenance schedulinr priority and
persistence-level operation priority.

vekterli committed:
Increase priority for global bucket merges

To avoid global buckets competing with (and usually being starved by)
default bucket space merges, explicitly prioritize default bucket
merges above most other load.

Increases both distributor-internal maintenance schedulinr priority and
persistence-level operation priority.

toregge commented on storage/src/vespa/storage/distributor/statecheckers.cpp:
```diff
@@ -861,11 +862,18 @@ SynchronizeAndMoveStateChecker::check(StateChecker::Context& c)
+            // Since the default bucket space has a dependency on the global bucket space,
+            // we prioritize scheduling of merges to global buckets over those for default buckets.
+            schedPri = MaintenancePriority::HIGH;
+            op->setPriority(c.distributorConfig.getMaintenancePriorities().mergeGlobalBuckets);
```
This might trigger another kind of starvation if new global documents are fed to a cluster where a node is full.

toregge replied:
After having looked at the maintenance scheduler, this doesn't seem to be a problem. Buckets are added to a bucket priority database by a maintenance scanner, and the scheduler just picks buckets from that bucket priority database. If a maintenance operation fails then it is not retried until the maintenance scanner checks the bucket again.

geirst reviewed (approved):
👍

vekterli committed:
Make prioritization of global bucket merges live configurable

Lets an application that is starving default space merges explicitly
disable prioritization temporarily. This is a stop-gap before we add
more robust and generalized handling of resource exhaustion scenarios.

vekterli commented:
@geirst @toregge PTAL at the latest commit. I've added an escape valve for manually getting out of a resource exhaustion scenario where global merges are starving default merges. This is of course not optimal, but serves as a stop-gap until we have more robust and generalized resource exhaustion handling across nodes.

geirst reviewed (approved):
👍
